coordinator/src/main/scala/filodb.coordinator/queryengine2/QueryEngine.scala

```diff
@@ -191,13 +195,42 @@ class QueryEngine(dataset: Dataset,
     // Translate column name to ID and validate here
     val colName = if (lp.column.isEmpty) dataset.options.valueColumn else lp.column
     val colID = dataset.colIDs(colName).get.head
-    shardsFromFilters(lp.filters, options).map { shard =>
+    val renamedFilters = renameMetricFilter(lp.filters)
+    shardsFromFilters(renamedFilters, options).map { shard =>
       val dispatcher = dispatcherForShard(shard)
       SelectChunkInfosExec(queryId, submitTime, options.itemLimit, dispatcher, dataset.ref, shard,
-        lp.filters, toRowKeyRange(lp.rangeSelector), colID)
+        renamedFilters, toRowKeyRange(lp.rangeSelector), colID)
     }
   }
 
+  /**
+   * Renames Prom AST __name__ metric name filters to one based on the actual metric column of the dataset,
+   * if it is not the prometheus standard
+   */
+  private def renameMetricFilter(filters: Seq[ColumnFilter]): Seq[ColumnFilter] =
```

Comment:
IMO, the rename is more appropriate in the Plan materialization time, rather than at execution time. Can we move this to the QueryEngine class ?

Reply:
I am mistaken! My eyes deceived me. For some reason, I thought this was in the SelectPartitionsExec class. :)